tests/integration/test_charm.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = METADATA["name"]
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test):
```

Comment:
Besides testing build and deployment, is it possible that we can also test the integration with a requirer charm?

Reply:
Will be addressed by PR #50